### DISCUSSION TITLE
Documentation says Provider is synchronous, yet Marvel Example uses asynchronous method with Provider

### DISCUSSION THREAD

**Discussion by h-unterp** [rrousselGit/riverpod#1454] [Q&A]
It says "Provider is a powerful tool for caching synchronous operations when combined with [ref.watch][(https://riverpod.dev/docs/concepts/reading#using-refwatch-to-observe-a-provider)." ](https://riverpod.dev/docs/providers/provider)

Yet the marvel example uses a Provider however the object it is providing has an async method:
https://github.com/rrousselGit/riverpod/blob/2f03ee04f3176dced8cccd27678df03c27970085/examples/marvel/lib/src/marvel.dart#L36

This is conflicting. Can you please clarify?

**Comment by rrousselGit** [comment]
I'm not understanding what your issue with `fetchCharacters` is.

This method isn't used by the provider. It's simply exposed for other providers to use
What exactly is conflicting here?

**Comment by h-unterp** [comment]
Bear in mind, I'm new to Riverpod.

"What exactly is conflicting here?" 
I'm even more confused now, some more clarification is needed:

"This method isn't used by the provider. It's simply exposed for other providers to use"

So what would be an example of a method that is used by a provider?  

Also, its not clear why he is using Provider for this pretty complicated MarvelRepository as https://riverpod.dev/docs/concepts/providers says that StateNotifier is for complex states.

**Reply by rrousselGit** [reply]
You can look at the marvel example to see how that fetchCharacters function is used

This provider doesn't do much. It's just creating an instance of a Repository. Kind of like flutter_bloc's RepositoryProvider if you're familiar with it?

I'd like to help you, but I'm having trouble understanding what your problem is.
